src/algebra/category/Module/basic.lean

```diff
@@ -35,10 +39,26 @@ instance : concrete_category (Module.{u} R) :=
   forget := { obj := λ R, R, map := λ R S f, (f : R → S) },
   forget_faithful := { } }
 
+/-- The object in the category of R-modules associated to an R-module -/
 def of (X : Type u) [add_comm_group X] [module R X] : Module R := ⟨R, X⟩
 
--- TODO: Once #1445 has merged, replace this with `has_zero_object (Module R)`
-instance : has_zero (Module R) := ⟨of R punit⟩
+instance : inhabited (Module R) := ⟨of R punit⟩
+
+lemma of_apply (X : Type u) [add_comm_group X] [module R X] : X = (of R X) := rfl
```

Comment:
Is there a reason you state the equality in this direction? Usually the "complicated" expression should go on the left.

Reply:
The reason was because the statement is actually  `X = ↥(of R X)`, so putting `X` first avoids explicit type annotations to get the correct coercion, but the explicit annotation makes it clearer what is going on, which is one more argument in favor of putting the `X` second. If will change it.